tempesta_fw/t/unit/test_http_parser.c

```diff
@@ -1673,6 +1673,118 @@ TEST(http_parser, empty_host)
 	FOR_REQ("GET / HTTP/1.1\n"
 		"Host:  \n"
 		"\n");
+
+	FOR_REQ("GET / HTTP/1.1\n"
+		"Host:    tempesta-tech.com   \n"
+		"\n")
+	{
+		TfwStr *host = &req->h_tbl->tbl[TFW_HTTP_HDR_HOST];
+		TfwStr h_expected = {
+			.chunks = (TfwStr []) {
+				{ .data = "Host:" , .len = 5 },
+				{ .data = "    " , .len = 4,
+				  .flags = TFW_STR_OWS },
+				{ .data = "tempesta-tech.com" , .len = 17,
+				  .flags = TFW_STR_VALUE },
+			},
+			.len = 26,
+			.nchunks = 3
+		};
+		test_string_split(&h_expected, host);
+
+		EXPECT_EQ(req->host_port, 0);
+	}
+
+	FOR_REQ("GET / HTTP/1.1\n"
+		"Host:    tempesta-tech.com:443   \n"
+		"\n")
+	{
+		TfwStr *host = &req->h_tbl->tbl[TFW_HTTP_HDR_HOST];
+		TfwStr h_expected = {
+			.chunks = (TfwStr []) {
+				{ .data = "Host:" , .len = 5 },
+				{ .data = "    " , .len = 4,
+				  .flags = TFW_STR_OWS },
+				{ .data = "tempesta-tech.com" , .len = 17,
+				  .flags = TFW_STR_VALUE },
+				{ .data = ":" , .len = 1 },
+				{ .data = "443" , .len = 3,
+				  .flags = TFW_STR_VALUE },
+			},
+			.len = 30,
+			.nchunks = 5
+		};
+		test_string_split(&h_expected, host);
+
+		EXPECT_EQ(req->host_port, 443);
+	}
+
+	FOR_REQ("GET / HTTP/1.1\n"
+		"Host:    [fd42:5ca1:e3a7::1000]   \n"
+		"\n")
+	{
+		TfwStr *host = &req->h_tbl->tbl[TFW_HTTP_HDR_HOST];
+		TfwStr h_expected = {
+			.chunks = (TfwStr []) {
+				{ .data = "Host:" , .len = 5 },
+				{ .data = "    " , .len = 4,
+				  .flags = TFW_STR_OWS },
+				{ .data = "[fd42:5ca1:e3a7::1000]" , .len = 22,
+				  .flags = TFW_STR_VALUE },
+			},
+			.len = 31,
+			.nchunks = 3
+		};
+		test_string_split(&h_expected, host);
+
+		EXPECT_EQ(req->host_port, 0);
+	}
+
+	FOR_REQ("GET / HTTP/1.1\n"
+		"Host:    [fd42:5ca1:e3a7::1000]:443   \n"
+		"\n")
+	{
+		TfwStr *host = &req->h_tbl->tbl[TFW_HTTP_HDR_HOST];
+		TfwStr h_expected = {
+			.chunks = (TfwStr []) {
+				{ .data = "Host:" , .len = 5 },
+				{ .data = "    " , .len = 4,
+				  .flags = TFW_STR_OWS },
+				{ .data = "[fd42:5ca1:e3a7::1000]" , .len = 22,
+				  .flags = TFW_STR_VALUE },
+				{ .data = ":" , .len = 1 },
+				{ .data = "443" , .len = 3,
+				  .flags = TFW_STR_VALUE },
+			},
+			.len = 35,
+			.nchunks = 5
+		};
+		test_string_split(&h_expected, host);
+
+		EXPECT_EQ(req->host_port, 443);
+	}
+
+	/* Port syntax is broken. */
+	EXPECT_BLOCK_REQ("GET / HTTP/1.1\n"
+			 "Host: tempesta-tech.com:443:1\n"
+			 "\n");
+	EXPECT_BLOCK_REQ("GET / HTTP/1.1\n"
+			 "Host: [fd42:5ca1:e3a7::1000]:443:1\n"
+			 "\n");
+	EXPECT_BLOCK_REQ("GET / HTTP/1.1\n"
+			 "Host: tempesta-tech.com::443\n"
+			 "\n");
+	EXPECT_BLOCK_REQ("GET / HTTP/1.1\n"
+			 "Host: tempesta-tech.com 443\n"
+			 "\n");
+
+	/* No braces around IPv6. */
```

Comment:
I believe that `[]` are called brackets. Braces are `{}'.

Reply:
Fixed.